In [1]:
# load sql extension
%load_ext sql

In [22]:
#connect to IO database
%sql mysql://USER:PASSWORD@SERVER/DATABASE

'Connected: admin@IO_analytics'

Test Connection to database

In [7]:
%%sql
SELECT *
FROM users
LIMIT 10;

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
10 rows affected.


[(1, 'test@test.com', None, 'Trails Testing', datetime.datetime(2020, 11, 20, 0, 1, 51), datetime.datetime(2020, 12, 17, 19, 26, 37), None),
 (3, 'tripstest@test.com', 'trips test', 'trips test', datetime.datetime(2020, 12, 1, 1, 52, 44), datetime.datetime(2020, 12, 16, 7, 0, 18), None),
 (4, 'dfoody@lion.lmu.edu', 'Dfoody', 'devon', datetime.datetime(2020, 12, 1, 21, 28, 31), datetime.datetime(2020, 12, 17, 1, 26, 7), 'I like camping with friends'),
 (5, 'bchen13@lion.lmu.edu', 'benltw', 'Ben', datetime.datetime(2020, 12, 1, 23, 15, 59), datetime.datetime(2021, 3, 9, 5, 51, 8), None),
 (10, 'jlach@lion.lmu.edu', 'JLach17', 'Julian Lach', datetime.datetime(2020, 12, 3, 1, 11, 35), datetime.datetime(2020, 12, 17, 19, 26, 38), 'Hi, I am Julian and I am from Los Angeles'),
 (11, 'apoirie1@lion.lmu.edu', None, 'Alexander Poirier', datetime.datetime(2020, 12, 3, 1, 13, 38), datetime.datetime(2020, 12, 17, 19, 23, 57), None),
 (12, 'mcolber1@lion.lmu.edu', 'katycolbert', 'Katy', datetime.datetime(2020, 12, 3, 5, 8, 23), datetime.datetime(2020, 12, 16, 7, 0, 19), None),
 (13, 'test@lion.lmu.edu', None, 'Max Test', datetime.datetime(2020, 12, 7, 0, 35, 12), datetime.datetime(2020, 12, 15, 5, 18, 26), None),
 (14, 'aritajustin@outlook.com', None, 'Justin', datetime.datetime(2020, 12, 8, 22, 10, 48), datetime.datetime(2020, 12, 8, 22, 10, 51), None),
 (15, 'gregory.lontok@lmu.edu', 'lontok ', 'Greg Lontok', datetime.datetime(2020, 12, 10, 19, 31, 47), datetime.datetime(2021, 4, 7, 1, 17, 21), None)]

Query #1 : Exploratory- How many trails were pulled from adalo. The goal of this query is to gain an understanding about how many trails I will be dealinig with for the entirety of this project.

In [10]:
%%sql
SELECT
    COUNT(*)
FROM trails

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
1 rows affected.


[(2977,)]

Just under 3000 total trails that were successfully pulled from adalo. This helps me to know I should have over 10,000 entries in the trips table in order to have all of the trails used and larger numbers on the number of trips per each trail. Only having 1000 entries caused a large amount of trails to remain unndiscovered by our users.

Query #2 - Distinct Locations of the trails collected. This query should provide us with a better understanding of where these trails are located so that we can better understand our users and where they are able to hike.

In [14]:
%%sql 
SELECT
    COUNT(DISTINCT(location)) AS number_of_distinct_hiking_locations
FROM trails;

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
1 rows affected.


[(370,)]

In [15]:
%%sql
SELECT 
    DISTINCT(location)
FROM trails;

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
370 rows affected.


[('Kenwood, California',),
 ('Inverness, California',),
 ('Bolinas, California',),
 ('Lagunitas-Forest Knolls, California',),
 ('Woodside, California',),
 ('Carson City, Nevada',),
 ('San Dimas, California',),
 ('San Antonio Heights, California',),
 ('Auburn Lake Trails, California',),
 ('Chester, California',),
 ('Susanville, California',),
 ('Westwood, California',),
 ('Alturas, California',),
 ('Johnstonville, California',),
 ('Hollywood, California',),
 ('Kingsbury, Nevada',),
 ('Thousand Oaks, California',),
 ('Truckee, California',),
 ('Carmel Valley Village, California',),
 ('Soledad, California',),
 ('Altadena, California',),
 ('Winters, California',),
 ('Los Osos, California',),
 ('Avocado Heights, California',),
 ('Downieville, California',),
 ('Soquel, California',),
 ('Auburn, California',),
 ('Sonoma, California',),
 ('Yosemite Valley, California',),
 ('Indian Hills, Nevada',),
 ('Hacienda Heights, California',),
 ('Moorpark, California',),
 ('Casa Conejo, California',),
 ('Littlerock, California',),
 ('Aliso Viejo, California',),
 ('Pine Mountain Club, California',),
 ('Saratoga, California',),
 ('Glendora, California',),
 ('Squaw Valley, California',),
 ('Three Rivers, California',),
 ('North Tustin, California',),
 ('Newport Beach, California',),
 ('Santa Rosa, California',),
 ('Costa Mesa, California',),
 ('Santa Monica, California',),
 ('Malibu, California',),
 ('Portola Valley, California',),
 ('Piedmont, California',),
 ('Buellton, California',),
 ('Badwater, California',),
 ('Tulelake, California',),
 ('Mill Valley, California',),
 ('Tamalpais Valley, California',),
 ('Wrightwood, California',),
 ('Mammoth Lakes, California',),
 ('South Lake Tahoe, California',),
 ('Marin City, California',),
 ('Tamalpais-Homestead Valley, California',),
 ('Sausalito, California',),
 ('Pacifica, California',),
 ('Millbrae, California',),
 ('Montara, California',),
 ('Tuolumne City, California',),
 ('Bridgeport, California',),
 ('San Francisco, California',),
 ('Big Pine, California',),
 ('Lone Pine, California',),
 ('Atascadero, California',),
 ('Topanga, California',),
 ('Mission Canyon, California',),
 ('Oakhurst, California',),
 ('Midpines, California',),
 ('Incline Village, Nevada',),
 ('Fontana, California',),
 ('Mojave, California',),
 ('Inyokern, California',),
 ('Quincy, California',),
 ('Burney, California',),
 ('Dunsmuir, California',),
 ('Weed, California',),
 ('Happy Camp, California',),
 ('Mogul, Nevada',),
 ('Sierra Madre, California',),
 ('Soulsbyville, California',),
 ('Isla Vista, California',),
 ('Solvang, California',),
 ('Santa Cruz Island, California',),
 ('Summerland, California',),
 ('Anacapa Island, California',),
 ('Danville, California',),
 ('Almaden Valley, California',),
 ('Rancho Cucamonga, California',),
 ('La Cañada Flintridge, California',),
 ('King City, California',),
 ('Roseland, California',),
 ('Dixon Lane-Meadow Creek, California',),
 ('Arcadia, California',),
 ('Ferndale, California',),
 ('Shasta, California',),
 ('Auberry, California',),
 ('Bertsch-Oceanview, California',),
 ('Diablo, California',),
 ('Monte Sereno, California',),
 ('Mount Hermon, California',),
 ('La Honda, California',),
 ('Clayton, California',),
 ('Blackhawk, California',),
 ('Santa Susana, California',),
 ('Avalon, California',),
 ('West Bishop, California',),
 ('Rohnert Park, California',),
 ('Half Moon Bay, California',),
 ('Tiburon, California',),
 ('San Luis Obispo, California',),
 ('Yreka, California',),
 ('Mount Shasta, California',),
 ('Twain Harte, California',),
 ('Newark, California',),
 ('Shingletown, California',),
 ('Villa Park, California',),
 ('Union City, California',),
 ('Hidden Hills, California',),
 ('Agoura, California',),
 ('Camarillo, California',),
 ('Loyola, California',),
 ('Meadow Vista, California',),
 ('Ashland, California',),
 ('Milpitas, California',),
 ('Fremont, California',),
 ('Alamo, California',),
 ('Ladera, California',),
 ('Meiners Oaks, California',),
 ('Stanford, Californi

found 370 distinct locations, as well as a list to look at the specific locations available to us. We can see that all of our locations come from two states: California and Nevada. This seems to be a limited list in terms of location and so as the app grows, we can take a look here to hopefully see an increase in the coverage that the app has so that individuals in states all over the country can use and enjoy the app.

Query #3 - Counting the Users available to us. This query is a simple, yet effective query that allows us to see our user base. We can see how many users we have to better understand how popular the app is and how often these idnividuals are using the app in order to plan an outdoor adventure.

In [16]:
%%sql
SELECT
    COUNT(*)
FROM users;

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
1 rows affected.


[(73,)]

In [18]:
%%sql
SELECT
    user_id,
    full_name,
    created_at
FROM users
ORDER BY created_at DESC
LIMIT 3;
    

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
3 rows affected.


[(81, 'bruno Aschidamini ', datetime.datetime(2021, 4, 28, 20, 8, 22)),
 (80, 'Patrick depeters', datetime.datetime(2021, 4, 28, 4, 22, 37)),
 (79, 'Will Bosques', datetime.datetime(2021, 4, 26, 5, 31, 4))]

Able to see we have 73 users that were pulled from Adalo, as well as the 3 most recent sign ups. These queries help us to understand the limitations of Inspire Outdoors at the moment. Hopefully, we will see this number increase in total users as well as having the most recent account creations being as close to real time as possible. The closer that these accounts are created to real time, the more successful the app is signifying that users are steadily signing up for the app.

Query #4 - Order users by # of trips planned (greatest to least). The goal of this query is to look at the trips that users are taking and seeing who our most active users of the app are. We will be able to see how many hikes they are going on and who they are from a full name perspective.

In [19]:
%%sql
SELECT
    COUNT(trips.user_id) AS number_of_hikes,
    users.full_name
FROM trips
JOIN users
    ON trips.user_id = users.user_id
GROUP BY trips.user_id
ORDER BY COUNT(trips.user_id) DESC;

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
41 rows affected.


[(33, 'Sprint Review'),
 (33, 'Will Bosques'),
 (32, 'Patrick depeters'),
 (32, 'John l Davis '),
 (30, 'Lola P'),
 (29, 'Jane Doe'),
 (29, 'Alec Flournoy'),
 (29, 'Bhupendra Gautam'),
 (29, 'Testertester'),
 (28, 'Google Play'),
 (28, 'Testing Notification24'),
 (27, 'Google@123'),
 (27, 'John'),
 (27, 'Miguel A.'),
 (26, 'User Example'),
 (26, 'Testing Notification'),
 (26, 'testapple'),
 (25, 'tester2'),
 (25, 'Alexander Poirier'),
 (25, 'Temp2021!'),
 (24, 'giancarlo'),
 (24, 'testing'),
 (24, 'Testing Notification22'),
 (24, 'John Smith'),
 (23, 'bruno Aschidamini '),
 (23, 'Olivia S'),
 (23, 'Carlos Anaya'),
 (23, 'terms page'),
 (23, 'Apple Review'),
 (22, 'Michael King'),
 (22, 'Nawaf Alkhuzaim'),
 (21, 'Benjamin Schneider-Haney'),
 (21, 'Sydney Toth'),
 (21, 'Ian Recio'),
 (19, 'yestttest '),
 (19, 'Testing Notification'),
 (18, 'Testing Notification'),
 (16, 'Bob Smith'),
 (15, 'Max Weber'),
 (15, 'Jane Doe'),
 (14, 'Testing Notification')]

Shows a list of how many hikes were taken by each individual. We can see our highest users were logging 30+ hikes. This helps us to see if users are returning to our app and enjoying the layout and features available. We could then reach out to these top and bottom users to hopefully get some feedback on how to improve the app in the future.

Query #5 - Most popular trails. This query will look and find which trails are frequently used by our users.

In [20]:
%%sql
SELECT
    COUNT(trips.trail_id) AS number_of_hikes,
    trails.trail_name,
    location,
    trails.length
FROM trips
JOIN trails
    ON trips.trail_id = trails.trail_id
GROUP BY trips.trail_id
ORDER BY COUNT(trips.trail_id) DESC;

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
842 rows affected.


[(4, "Kid's Trail", 'Yreka, California', 1),
 (4, 'Lakes Trail', 'Three Rivers, California', 0),
 (3, 'Big Five Lakes Trail', 'Three Rivers, California', 4),
 (3, 'Middle Fork Kaweah Connector', 'Three Rivers, California', 1),
 (3, 'Shotgun Pass', 'Three Rivers, California', 4),
 (3, 'Kirk Creek Trail #4E17', 'King City, California', 5),
 (3, 'Fern Creek Trail', 'Mill Valley, California', 1),
 (3, 'Ridge Trail', 'Gilroy, California', 2),
 (3, 'Lower Palos Colorados', 'Piedmont, California', 1),
 (3, 'Charmlee Wilderness Park Loop', 'Malibu, California', 2),
 (3, 'Crystal Lake Trail', 'Mammoth Lakes, California', 0),
 (3, 'Garden Gulch Trail', 'Fort Jones, California', 9),
 (3, 'Cone Peak Trail #4E12', 'King City, California', 3),
 (3, 'Beach Access Trail', 'Marina, California', 0),
 (3, 'Salt Creek Interpretive Trail', 'Badwater, California', 1),
 (2, 'Ash Canyon to Kings Canyon Trail', 'Carson City, Nevada', 7),
 (2, 'The Wall - a steep connector.', 'Susanville, California', 0),
 (2, 'Dairy Trail', 'Susanville, California', 0),
 (2, 'The Jagged Edge', 'Susanville, California', 0),
 (2, 'Beach Trail', 'Carmel Valley Village, California', 1),
 (2, 'Bear Gulch Cave Trail', 'Soledad, California', 0),
 (2, 'Connector Trail - Connects Chaparral Picnic Area to Juniper Canyon Trail.', 'Soledad, California', 0),
 (2, 'Zumwalt Meadows Loop Trail', 'Squaw Valley, California', 1),
 (2, 'Mist Falls', 'Squaw Valley, California', 8),
 (2, 'Sullivan Fire Road', 'Santa Monica, California', 6),
 (2, 'Wunderlich Loop', 'Woodside, California', 10),
 (2, 'Marble Falls', 'Three Rivers, California', 7),
 (2, 'High Sierra Camp: May Lake- Glen Aulin- Tuolumne Meadows', 'Yosemite Valley, California', 20),
 (2, 'California Coastal Trail: Section 28 & 29', 'Bolinas, California', 5),
 (2, 'The Hub Out-and-Back', 'Santa Monica, California', 16),
 (2, 'Laurel Lake Trail', 'Tuolumne City, California', 5),
 (2, 'Alta-High Sierra Cut-off', 'Three Rivers, California', 2),
 (2, 'Crystal Lakes', 'Three Rivers, California', 2),
 (2, 'Hotel Creek Overlook Trail', 'Squaw Valley, California', 0),
 (2, 'Mosquito Lakes Trail', 'Three Rivers, California', 4),
 (2, 'Tokopah Falls Trail', 'Three Rivers, California', 2),
 (2, 'Tilden Creek - Bailey Ridge Trail', 'Yosemite Valley, California', 5),
 (2, 'Lembert Dome', 'Yosemite Valley, California', 2),
 (2, 'Lobo Canyon Trail', 'Isla Vista, California', 2),
 (2, 'Middle Fork Trail #6W01', 'Wrightwood, California', 5),
 (2, 'Juniper to Summit Loop', 'Diablo, California', 7),
 (2, 'West End Road', 'Avalon, California', 12),
 (2, 'Joe Crane Lake Trail', 'Mammoth Lakes, California', 1),
 (2, 'Bathtub Lake Trail', 'Chester, California', 1),
 (2, 'Juniper Lake to Snag Lake', 'Chester, California', 3),
 (2, 'Manzanita Lake Loop-Campground Connector Trail', 'Shingletown, California', 0),
 (2, 'Lake Chabot Inner Loop', 'Ashland, California', 9),
 (2, 'Peak Trail', 'Fremont, California', 4),
 (2, 'Gene Marshall Out and Back', 'Meiners Oaks, California', 14),
 (2, 'Atkinson Bluff Trail', 'Boulder Creek, California', 2),
 (2, 'North Fork Lone Pine Creek Trail', 'Lone Pine, California', 3),
 (2, 'Overlook Trail', 'Bertsch-Oceanview, California', 0),
 (2, 'Rhododendron Trail', 'Westhaven-Moonstone, California', 8),
 (2, 'Trillium Falls Trail', 'Westhaven-Moonstone, California', 2),
 (2, 'McArthur Creek Loop Spur', 'Westhaven-Moonstone, California', 0),
 (2, 'Mill Creek Horse Trail (Alt Loop)', 'Bertsch-Oceanview, California', 0),
 (2, 'Alder Basin Trail', 'Bertsch-Oceanview, California', 1),
 (2, 'Dry Lagoon Campground Trail', 'Westhaven-Moonstone, California', 0),
 (2, 'Barlow Canyon Trail', 'Salinas, California', 1),
 (2, 'Arcata Community Forest Tour', 'Arcata, California', 5),
 (2, 'California Trail #5', 'Arcata, California', 1),
 (2, 'Vista Trail #10', 'Arcata, California', 1),
 (2, 'East Shore Trail', 'Castro Valley, California', 2),
 (2, 'Bayview Heights to Broderson', 'Los Osos, California', 1),
 (2, 'South Gerstle Trail', 'Sea 

842 were hiked this year, meaning almost 2,000 of our trails remained untraveled by our users. This is mainly due to the limited amount of trips we currently have logged. The results help to show how users were extremely active in Three Rivers California, which could be an area we could look towards to promote our app and establish a community there, sponser some of the trails in the area, or ask those individuals for feedback to see why they like using our app at Three Rivers.

------------------------------------------------------------------------------------------------------

Main Business Question: When looking at the summertime, how many hikes were taken by users? Compare these resuults to the fall.

Business Justification: Knowing that summer is a popular time for individuals to be outdoors and active, we as a company will want to look durinig peak season and see how many hikes we have logged. This will help us to see if as a company we are succeeding during a season that we expect volume to be at its highest.

Features: VIEW, JOIN, CTE

In [27]:
%%sql
CREATE VIEW summer_hikes AS
SELECT 
    trip_id,
    trips.date,
    trips.user_id,
    full_name,
    trips.trail_id,
    trail_name,
    location,
    trails.length
FROM trips
JOIN users
    ON users.user_id = trips.user_id
JOIN trails 
    ON trails.trail_id = trips.trail_id
WHERE 
    trips.date BETWEEN '2020-05-10' AND '2020-08-10';

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
0 rows affected.


[]

In [28]:
%%sql
SELECT *
FROM summer_hikes;

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
392 rows affected.


[(1, datetime.datetime(2020, 5, 29, 0, 0), 48, 'Alec Flournoy', 7002827, 'Connector 1 - a fun, narrow connector.', 'Susanville, California', 0),
 (7, datetime.datetime(2020, 6, 27, 0, 0), 76, 'Lola P', 7089771, 'Foresthill Divide Loop', 'Auburn Lake Trails, California', 11),
 (8, datetime.datetime(2020, 6, 10, 0, 0), 63, 'Google Play', 7009704, 'Tilden Creek - Bailey Ridge Trail', 'Yosemite Valley, California', 5),
 (11, datetime.datetime(2020, 8, 10, 0, 0), 80, 'Patrick depeters', 7021573, 'Shooters', 'San Luis Obispo, California', 1),
 (15, datetime.datetime(2020, 6, 30, 0, 0), 74, 'John l Davis ', 7005249, 'Bear Gulch Cave Trail', 'Soledad, California', 0),
 (18, datetime.datetime(2020, 8, 7, 0, 0), 47, 'Jane Doe', 7018011, 'Snow Creek Falls Trail', 'Yosemite Valley, California', 2),
 (19, datetime.datetime(2020, 7, 20, 0, 0), 63, 'Google Play', 7016774, 'Tennessee Valley Lollipop', 'Tamalpais-Homestead Valley, California', 4),
 (20, datetime.datetime(2020, 8, 2, 0, 0), 74, 'John l Davis ', 7009715, 'Rancheria Falls to Tiltill Valley', 'Yosemite Valley, California', 2),
 (22, datetime.datetime(2020, 7, 29, 0, 0), 62, 'Sprint Review', 7036115, 'Mayfair Ranch Trail', 'Almaden Valley, California', 3),
 (26, datetime.datetime(2020, 5, 11, 0, 0), 76, 'Lola P', 7020217, 'Horsetail Falls', 'South Lake Tahoe, California', 3),
 (29, datetime.datetime(2020, 5, 25, 0, 0), 46, 'Jane Doe', 7013806, 'Joe Crane Lake Trail', 'Mammoth Lakes, California', 1),
 (31, datetime.datetime(2020, 5, 14, 0, 0), 73, 'Bhupendra Gautam', 7058453, 'Sherwin and Valentine Lakes Loop', 'Mammoth Lakes, California', 12),
 (33, datetime.datetime(2020, 7, 30, 0, 0), 58, 'tester2', 7084144, 'Trail 52', 'Marina, California', 1),
 (50, datetime.datetime(2020, 6, 2, 0, 0), 48, 'Alec Flournoy', 7024830, 'Mt. Shasta Mine Loop Connector', 'Shasta, California', 0),
 (51, datetime.datetime(2020, 6, 19, 0, 0), 49, 'Testing Notification', 7022066, 'Le Mesnager Loop Trail', 'La Crescenta-Montrose, California', 1),
 (53, datetime.datetime(2020, 7, 4, 0, 0), 44, 'John', 7005840, 'Wunderlich Loop', 'Woodside, California', 10),
 (59, datetime.datetime(2020, 7, 9, 0, 0), 60, 'terms page', 7054625, 'Shell Beach Bluff Trail', 'Avila Beach, California', 1),
 (60, datetime.datetime(2020, 5, 30, 0, 0), 59, 'Apple Review', 7042275, 'Ridgeline Trail', 'Livermore, California', 4),
 (64, datetime.datetime(2020, 5, 28, 0, 0), 45, 'John Smith', 7089948, 'Culvert Trail', 'North Auburn, California', 1),
 (66, datetime.datetime(2020, 5, 13, 0, 0), 72, 'Miguel A.', 7030899, 'Telegraph Peak Trail #7W06A', 'Wrightwood, California', 0),
 (69, datetime.datetime(2020, 6, 19, 0, 0), 53, 'Testing Notification22', 7032092, '35E311', 'Big Pine, California', 3),
 (70, datetime.datetime(2020, 5, 31, 0, 0), 74, 'John l Davis ', 7005687, 'Sullivan Fire Road', 'Santa Monica, California', 6),
 (72, datetime.datetime(2020, 6, 18, 0, 0), 48, 'Alec Flournoy', 7024571, 'Montara Mountain from Gray Whale Cove', 'Montara, California', 10),
 (76, datetime.datetime(2020, 5, 20, 0, 0), 61, 'yestttest ', 7016604, 'Marble Canyon Road', 'Lone Pine, California', 3),
 (77, datetime.datetime(2020, 7, 11, 0, 0), 75, 'Max Weber', 7016225, 'Mori Point', 'Pacifica, California', 2),
 (78, datetime.datetime(2020, 5, 16, 0, 0), 79, 'Will Bosques', 7014110, 'Manzanita Lake Loop-Campground Connector Trail', 'Shingletown, California', 0),
 (83, datetime.datetime(2020, 8, 2, 0, 0), 55, 'testing', 7077302, 'Oak Woodland Trail', 'San Luis Obispo, California', 1),
 (88, datetime.datetime(2020, 7, 4, 0, 0), 45, 'John Smith', 7005922, 'Marble Falls', 'Three Rivers, California', 7),
 (92, datetime.datetime(2020, 5, 25, 0, 0), 61, 'yestttest ', 7020495, 'Rubicon Peak', 'Tahoma, California', 4),
 (96, datetime.datetime(2020, 6, 15, 0, 0), 74, 'John l Davis ', 7014133, 'Widow Lake to Snag Lake Trail', 'Chester, California', 9),
 (97, datetime.datetime(2020, 6, 9, 0, 0), 53, 'Testing Notification22', 7033021, 'Cucamonga Peak via Icehouse Canyo

Analysis: There were 392 hikes that were logged for our users between May 10 and August 10 of 2020.

SQL below to compare summer to the fall hiking to see if there is a difference

In [29]:
%%sql
WITH fallHikesCTE AS (
   SELECT 
        trip_id,
        trips.date,
        trips.user_id,
        full_name,
        trips.trail_id,
        trail_name,
        location,
        trails.length
    FROM trips
    JOIN users
        ON users.user_id = trips.user_id
    JOIN trails 
        ON trails.trail_id = trips.trail_id
    WHERE 
        trips.date BETWEEN '2020-08-15' AND '2020-11-15'
)
SELECT *
FROM fallHikesCTE;

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
391 rows affected.


[(2, datetime.datetime(2020, 9, 14, 0, 0), 58, 'tester2', 7014067, 'Butte Lake to Snag Lake Trail (East)', 'Chester, California', 7),
 (3, datetime.datetime(2020, 10, 23, 0, 0), 55, 'testing', 7006581, 'Table Rock 27k', 'Bolinas, California', 17),
 (4, datetime.datetime(2020, 11, 5, 0, 0), 68, 'Benjamin Schneider-Haney', 7024991, 'Iron Horse Regional Trail: Stone Valley Rd - Sycamore Valley Rd', 'Alamo, California', 4),
 (5, datetime.datetime(2020, 11, 8, 0, 0), 52, 'Testing Notification', 7012889, 'Sentinel Dome Trail', 'Yosemite Valley, California', 1),
 (9, datetime.datetime(2020, 9, 2, 0, 0), 49, 'Testing Notification', 7014395, 'Cheeseboro Ridge & Cheeseboro Canyon Trail', 'Agoura, California', 10),
 (16, datetime.datetime(2020, 9, 7, 0, 0), 69, 'Sydney Toth', 7025648, 'Elevator Trail', 'San Luis Obispo, California', 1),
 (17, datetime.datetime(2020, 10, 10, 0, 0), 79, 'Will Bosques', 7066545, 'Old Cañon Trail', 'Piedmont, California', 1),
 (21, datetime.datetime(2020, 10, 26, 0, 0), 71, 'Olivia S', 7097697, 'Drury-Chaney Trail', 'Rio Dell, California', 2),
 (24, datetime.datetime(2020, 10, 15, 0, 0), 79, 'Will Bosques', 7016219, 'Sweeny Ridge Loop', 'Pacifica, California', 6),
 (25, datetime.datetime(2020, 8, 25, 0, 0), 43, 'Alexander Poirier', 7088655, 'Stony Creek Trail', 'Three Rivers, California', 4),
 (30, datetime.datetime(2020, 10, 17, 0, 0), 64, 'Google@123', 7017240, 'Trillium Falls Trail', 'Westhaven-Moonstone, California', 2),
 (32, datetime.datetime(2020, 9, 17, 0, 0), 49, 'Testing Notification', 7029867, 'Trinidad Head Weather Station Spur', 'Westhaven-Moonstone, California', 0),
 (34, datetime.datetime(2020, 9, 28, 0, 0), 47, 'Jane Doe', 7002763, 'Bizz Johnson National Recreation Trail', 'Westwood, California', 26),
 (36, datetime.datetime(2020, 10, 1, 0, 0), 49, 'Testing Notification', 7033628, 'Gridley Trail to Nordhoff Peak', 'Ojai, California', 14),
 (38, datetime.datetime(2020, 10, 11, 0, 0), 80, 'Patrick depeters', 7014877, 'Topanga Creek Outpost Loop', 'Topanga, California', 11),
 (39, datetime.datetime(2020, 9, 23, 0, 0), 78, 'Bob Smith', 7024380, 'Mark West Creek Trail', 'Larkfield-Wikiup, California', 1),
 (43, datetime.datetime(2020, 9, 16, 0, 0), 53, 'Testing Notification22', 7007547, 'Laurel Lake Trail', 'Tuolumne City, California', 5),
 (45, datetime.datetime(2020, 9, 22, 0, 0), 72, 'Miguel A.', 7008496, 'Shotgun Pass', 'Three Rivers, California', 4),
 (46, datetime.datetime(2020, 9, 26, 0, 0), 73, 'Bhupendra Gautam', 7011110, 'Dawson Saddle to Mount Baden-Powell', 'Wrightwood, California', 9),
 (47, datetime.datetime(2020, 10, 31, 0, 0), 48, 'Alec Flournoy', 7032233, 'Donner State Park Tour', 'Truckee, California', 3),
 (49, datetime.datetime(2020, 9, 2, 0, 0), 80, 'Patrick depeters', 7049218, 'Beith Creek Loop', 'Bayside, California', 3),
 (52, datetime.datetime(2020, 9, 30, 0, 0), 49, 'Testing Notification', 7017321, 'Mill Creek Horse Trail (Alt Loop)', 'Bertsch-Oceanview, California', 0),
 (54, datetime.datetime(2020, 10, 7, 0, 0), 55, 'testing', 7032959, 'Temescal Canyon Loop', 'Santa Monica, California', 4),
 (56, datetime.datetime(2020, 9, 2, 0, 0), 47, 'Jane Doe', 7018041, 'Iron Creek Trail', 'Mammoth Lakes, California', 8),
 (57, datetime.datetime(2020, 9, 5, 0, 0), 70, 'Temp2021!', 7016865, 'Orcutt East Loop', 'Orcutt, California', 0),
 (61, datetime.datetime(2020, 10, 13, 0, 0), 65, 'Ian Recio', 7007643, 'Sutro Heights Loop Trail', 'San Francisco, California', 0),
 (65, datetime.datetime(2020, 11, 9, 0, 0), 56, 'giancarlo', 7023403, 'Clouds Rest - Meadow Trail Loop', 'La Honda, California', 2),
 (67, datetime.datetime(2020, 8, 19, 0, 0), 65, 'Ian Recio', 7015348, 'Peak Trail', 'Fremont, California', 4),
 (71, datetime.datetime(2020, 11, 12, 0, 0), 74, 'John l Davis ', 7037030, 'Snow Mountain Summits', 'Lucerne, California', 10),
 (74, datetime.datetime(2020, 10, 27, 0, 0), 81, 'bruno Aschidamini ', 7047282, 'Mammoth Rock Trail Connector', 'Mammoth Lakes, California', 1),
 (75, datet

When comparing the two segments, we see almost an identical number of hikes occuring in both the summer and fall. This can be explained due to thee consistency of weather in Calfornia. Other states often have larger ranges in terms or weather and temperature where the users here on the west coast are used to sunny and warm days year round, especially in the south. Since this is a set of mock data, it makes sense to see this since thee date was just a random variable, but we may  be surprised to see similar real life results to this due to the weather and active culture found on the west coast. There is a lot of good hiking year round in California, and we can expect our users to take advantage of it.

Business Query #2 - Looking in summer_hikes, lets rank based on location to see if we can find some popular trails

Business Justification: this kind of query will help Inspire Outdoors to locate areas of popularity. We can see where our users hike, how many trips are being logged there (regardless of the individual) and hopefully we can continue to see growth in all trails withmore users signinig up and more hikes being logged. 

Features: WINDOW, GROUP BY

In [33]:
%%sql
SELECT
    trip_id,
    trail_name,
    location,
    COUNT(trip_id) AS number_of_trips,
    DENSE_RANK() OVER (
        PARTITION BY location
        ORDER BY COUNT(trip_id) DESC
    ) AS trip_count_location_rank
FROM summer_hikes
GROUP BY trail_id;


 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
370 rows affected.


[(243, 'Talepop Loop', 'Agoura, California', 1, 1),
 (810, 'King Gillete Ranch Loop', 'Agoura, California', 1, 1),
 (717, 'Summit Ridge Trail', 'Alhambra, California', 1, 1),
 (550, 'Aliso Wood Canyon 20-Mile Loop', 'Aliso Viejo, California', 1, 1),
 (22, 'Mayfair Ranch Trail', 'Almaden Valley, California', 1, 1),
 (298, 'Woods Trail', 'Almaden Valley, California', 1, 1),
 (248, 'Bear Canyon Trail 12W08', 'Altadena, California', 1, 1),
 (693, 'Colby Canyon Trail #12W23', 'Altadena, California', 1, 1),
 (181, 'Pine Creek Trail #15E20', 'Alturas, California', 1, 1),
 (541, 'Inspiration Point', 'Anacapa Island, California', 1, 1),
 (846, 'Cathedral Cove', 'Anacapa Island, California', 1, 1),
 (999, 'Santa Cruz Traverse', 'Anacapa Island, California', 1, 1),
 (732, 'Natural Bridges Trail', 'Angels Camp, California', 1, 1),
 (175, 'Buggy Trail', 'Aptos, California', 1, 1),
 (273, 'West Ridge Trail', 'Aptos, California', 1, 1),
 (977, 'Terrace Trail', 'Aptos, California', 1, 1),
 (317, '14th St. - Redwood Park Connector', 'Arcata, California', 1, 1),
 (727, 'Arcata Community Forest Tour', 'Arcata, California', 1, 1),
 (578, 'Rancheria Falls', 'Auberry, California', 1, 1),
 (844, 'Blackcap Trail 29E03', 'Auberry, California', 1, 1),
 (7, 'Foresthill Divide Loop', 'Auburn Lake Trails, California', 1, 1),
 (681, 'Lakeview Connector Trail', 'Auburn Lake Trails, California', 1, 1),
 (756, 'Auburn Overlook Park Western States-Quarry Trails Loop (20 miles)', 'Auburn, California', 1, 1),
 (59, 'Shell Beach Bluff Trail', 'Avila Beach, California', 1, 1),
 (613, 'Pecho Coast Trail', 'Avila Beach, California', 1, 1),
 (287, 'Natural Bridge Trail', 'Badwater, California', 1, 1),
 (533, 'Mount Perry Trail', 'Badwater, California', 1, 1),
 (660, "Dante's View Trail", 'Badwater, California', 1, 1),
 (705, 'Funeral Slot Canyon Spur', 'Badwater, California', 1, 1),
 (993, 'Salt Creek Interpretive Trail', 'Badwater, California', 1, 1),
 (195, 'Overlook Trail', 'Bertsch-Oceanview, California', 2, 1),
 (313, 'Crescent Beach Trail (Spur)', 'Bertsch-Oceanview, California', 1, 2),
 (521, 'Alder Basin Trail', 'Bertsch-Oceanview, California', 1, 2),
 (673, 'Reservoir Road (Mill Creek)', 'Bertsch-Oceanview, California', 1, 2),
 (767, 'Coastal Trail Spur', 'Bertsch-Oceanview, California', 1, 2),
 (847, 'Hobbs Wall Trail', 'Bertsch-Oceanview, California', 1, 2),
 (69, '35E311', 'Big Pine, California', 1, 1),
 (211, 'Marie Louise Lakes Trail', 'Big Pine, California', 1, 1),
 (412, 'Moonlight Lake Trail', 'Big Pine, California', 1, 1),
 (485, 'Ubehebe Crater Loop', 'Big Pine, California', 1, 1),
 (595, 'Baker Lakes Trail', 'Big Pine, California', 1, 1),
 (652, 'State Lakes Loop', 'Big Pine, California', 1, 1),
 (751, 'Blackcap Pass Route', 'Big Pine, California', 1, 1),
 (919, 'Bench Lake Trail', 'Big Pine, California', 1, 1),
 (936, 'Simpson Meadow Trail', 'Big Pine, California', 1, 1),
 (312, 'Tassajara Creek Trail', 'Blackhawk, California', 1, 1),
 (395, 'California Coastal Trail: Section 28 & 29', 'Bolinas, California', 1, 1),
 (453, 'Rabbit Chase 30K', 'Bolinas, California', 1, 1),
 (697, 'Alamere Falls Trail', 'Bolinas, California', 1, 1),
 (258, 'Point Arena-Stornetta Public Lands Trail', 'Boonville, California', 1, 1),
 (647, 'Bigger Basin Loop', 'Boulder Creek, California', 1, 1),
 (222, 'Horse Meadow Trail', 'Bridgeport, California', 1, 1),
 (530, 'Mount Dana Trail', 'Bridgeport, California', 1, 1),
 (532, 'Stough Canyon Fire Road', 'Burbank, California', 1, 1),
 (99, 'Lower Oat Hill Mine Trail', 'Calistoga, California', 2, 1),
 (529, 'Chumash Trail', 'Camarillo, California', 1, 1),
 (104, 'Horseshoe', 'Carmel Valley Village, California', 1, 1),
 (125, 'Appaloosa Trail', 'Carmel Valley Village, California', 1, 1),
 (196, 'Waterfall Overlook Trail', 'Carmel Valley Village, California', 1, 1),
 (496, 'Beach Trail', 'Carmel Valley Village, California', 1, 1),
 (644, 'Pine Ridge Trail #3E06', 'Carmel Valley Village, California', 1, 1),
 (702, 'Trail Camp B

We can see majoirty of trails were only hiked once and can be in a tie for the most hiked given that specific location. Thiis is due to the limited number of entries we have for our trips table. Hoopefully with more data, we will be able to see larger numbers to signify enjoyment of our app from all different locations. One important note is we see that Three Rivers is a very popular area, and has over 10 trails that are in the top 2 most hiked. We can focus on Three Rivers in the next query to gain insight into how users feel about the area.

Business Query #3 - We found in the previous Query that Three Rivers California is a popular spot to hike with over 20 hikes occuring there this summer, lets take a look at all of the hikes that were logged at Three Rivers, and seeing how their ratings are. This query will return the trail name, a rating, the user who went on the trip, and finally an additional piece added with a CASE statment to determine if the review was bad, ok, or great.

Business Justification: this query helps us to see how users are responding to this popular area. We are hoping to see positive reviews, as this indicates customer satisfaction and will likely lead to them returning to the app in order to add new hikes in the future. If we see negative reviews, we can hoopefully reach out to these users to see if we can figure out improvements

Features: CASE, Subquery

In [37]:
%%sql
 SELECT
   	users.full_name,
    trails.trail_name,
    location, 
    trips.review,
    CASE
        WHEN trips.review = 0 OR trips.review = 1 THEN 'Poor'
        WHEN trips.review = 2 OR trips.review = 3 THEN 'Good'
        WHEN trips.review = 4 OR trips.review = 5 THEN 'Excellent'
    END AS user_experience
FROM trips
JOIN users
    ON users.user_id = trips.user_id
JOIN trails 
    ON trails.trail_id = trips.trail_id
WHERE location IN (
	SELECT 
		location
	FROM trails 
	WHERE location LIKE '%Three Rivers%'
)
ORDER BY user_experience;

 * mysql://admin:***@lmu-dev-01.cdfi2extn64i.us-east-1.rds.amazonaws.com/IO_analytics
61 rows affected.


[('Patrick depeters', 'Lakes Trail', 'Three Rivers, California', 4, 'Excellent'),
 ('Testertester', 'North Fork Trail', 'Three Rivers, California', 4, 'Excellent'),
 ('Nawaf Alkhuzaim', 'Potwisha-Hospital Rock', 'Three Rivers, California', 5, 'Excellent'),
 ('Testertester', 'Twin Lakes Trail', 'Three Rivers, California', 4, 'Excellent'),
 ('yestttest ', 'Ash Mtn River Trail', 'Three Rivers, California', 5, 'Excellent'),
 ('Google Play', 'Crystal Lakes', 'Three Rivers, California', 4, 'Excellent'),
 ('John l Davis ', 'Jennie Lake Trail', 'Three Rivers, California', 5, 'Excellent'),
 ('Testing Notification', 'Big Trees Trail', 'Three Rivers, California', 4, 'Excellent'),
 ('Alexander Poirier', 'Alta Peak/ Lakes Trail', 'Three Rivers, California', 4, 'Excellent'),
 ('yestttest ', 'Tokopah Falls Trail', 'Three Rivers, California', 4, 'Excellent'),
 ('Miguel A.', 'Redwood Meadow/Bearpaw', 'Three Rivers, California', 4, 'Excellent'),
 ('bruno Aschidamini ', 'Dorst/Lost Grove Access', 'Three Rivers, California', 5, 'Excellent'),
 ('Patrick depeters', 'Cold Springs Nature Trail Cutoff', 'Three Rivers, California', 4, 'Excellent'),
 ('tester2', 'Alta-High Sierra Cut-off', 'Three Rivers, California', 5, 'Excellent'),
 ('Google@123', 'Lakes Trail', 'Three Rivers, California', 5, 'Excellent'),
 ('Lola P', 'Tokopah Falls Trail', 'Three Rivers, California', 4, 'Excellent'),
 ('Testing Notification', 'Summit Lake', 'Three Rivers, California', 2, 'Good'),
 ('Alexander Poirier', 'Stony Creek Trail', 'Three Rivers, California', 2, 'Good'),
 ('Miguel A.', 'Shotgun Pass', 'Three Rivers, California', 3, 'Good'),
 ('John Smith', 'Marble Falls', 'Three Rivers, California', 3, 'Good'),
 ('Google@123', 'Sugar Pine Trail', 'Three Rivers, California', 2, 'Good'),
 ('John l Davis ', 'Mosquito Lakes Trail', 'Three Rivers, California', 3, 'Good'),
 ('tester2', 'Stony Creek Trail', 'Three Rivers, California', 3, 'Good'),
 ('Temp2021!', 'Bear Hill Trail', 'Three Rivers, California', 2, 'Good'),
 ('Testing Notification24', 'Viewpoint Spur', 'Three Rivers, California', 3, 'Good'),
 ('Carlos Anaya', 'Ladybug Trail', 'Three Rivers, California', 3, 'Good'),
 ('testing', 'Lakes Trail', 'Three Rivers, California', 3, 'Good'),
 ('Google@123', 'Lodgepole-Sherman Tree', 'Three Rivers, California', 3, 'Good'),
 ('bruno Aschidamini ', 'Hazelwood Trail', 'Three Rivers, California', 2, 'Good'),
 ('John', 'White Chief Spur', 'Three Rivers, California', 3, 'Good'),
 ('Max Weber', 'Fox Meadow Trailhead to Weaver Lake', 'Three Rivers, California', 2, 'Good'),
 ('Carlos Anaya', 'Sawtooth Pass Trail', 'Three Rivers, California', 2, 'Good'),
 ('yestttest ', 'Colonel Young Tree Trail', 'Three Rivers, California', 2, 'Good'),
 ('Benjamin Schneider-Haney', 'Middle Fork Kaweah Connector', 'Three Rivers, California', 2, 'Good'),
 ('User Example', 'Crystal Lakes', 'Three Rivers, California', 3, 'Good'),
 ('Will Bosques', 'Dorst/Lost Grove', 'Three Rivers, California', 2, 'Good'),
 ('Temp2021!', 'Farewell Gap Trail', 'Three Rivers, California', 2, 'Good'),
 ('Google@123', 'Ranger Lake Trail', 'Three Rivers, California', 3, 'Good'),
 ('terms page', 'Old Colony Mill Rd', 'Three Rivers, California', 2, 'Good'),
 ('Alexander Poirier', 'Lakes Trail', 'Three Rivers, California', 3, 'Good'),
 ('Will Bosques', 'Shotgun Pass', 'Three Rivers, California', 2, 'Good'),
 ('Testing Notification22', 'Washington Tree', 'Three Rivers, California', 2, 'Good'),
 ('Jane Doe', 'Rattlesnake', 'Three Rivers, California', 2, 'Good'),
 ('Google Play', 'Alta-High Sierra Cut-off', 'Three Rivers, California', 2, 'Good'),
 ('Testertester', 'Big Five Lakes Trail', 'Three Rivers, California', 2, 'Good'),
 ('John l Davis ', 'Middle Fork Kaweah Connector', 'Three Rivers, California', 3, 'Good'),
 ('Lola P', 'Shotgun Pass', 'Three Rivers, California', 1, 'Poor'),
 ('User Example', 'Middle Fork Kaweah Connector', 'Three Rivers, California', 0, 'Poor'),
 ('Sydney Toth', 'Marble Falls', 'Three Rivers, California', 1, 'Poo

Analysis: When looking at the results of the querey, it seems that most of the trips that were logged at Three Rivers were pretty positive. There were a handful of Poor and Excellent but majoirty of people rated their adventure in our Good category. This lets us see that users enjoyed their time at Three Rivers and we can hopefully assume that they will return to the area.

The Trigger and Stored Procedure are discussed about in their SQL documents.